assets/src/settings-page/plugin-suppression.js

```diff
@@ -1,45 +1,394 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
 
 /**
  * WordPress dependencies
  */
-import { useContext, useEffect, useRef, useState } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
+import { useContext, useEffect, useState, useRef, Fragment } from '@wordpress/element';
+import { __, sprintf, _n } from '@wordpress/i18n';
+import { autop } from '@wordpress/autop';
+import { format, dateI18n } from '@wordpress/date';
+import apiFetch from '@wordpress/api-fetch';
+import { SelectControl } from '@wordpress/components';
+import { addQueryArgs } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
 import { Options } from '../components/options-context-provider';
+import { ConditionalDetails } from '../components/conditional-details';
 import { Selectable } from '../components/selectable';
 
 /**
- * Plugin suppression section of the settings page.
+ * Renders the formatted date for when a plugin was suppressed.
+ *
+ * @param {Object} props Component props.
+ * @param {Object} props.suppressedPlugin
  */
-export function PluginSuppression() {
-	const { fetchingOptions } = useContext( Options );
+function SuppressedPluginTime( { suppressedPlugin } ) {
+	const { editedOptions } = useContext( Options );
+	const { date_format: dateFormat } = editedOptions;
 
-	const [ supportedTemplatesTableFound, setSupportedTemplatesTableFound ] = useState( false );
+	if ( ! suppressedPlugin || ! suppressedPlugin.timestamp || ! dateFormat ) {
+		return null;
+	}
 
-	const pluginSuppressionContainer = useRef();
+	return (
+		<time dateTime={ format( 'c', suppressedPlugin.timestamp ) }>
+			{
+				// Translators: placeholder is a formatted date.
+				sprintf( __( 'Since %s.', 'amp' ), dateI18n( dateFormat, suppressedPlugin.timestamp * 1000 ) )
+			}
+			{ ' ' }
+		</time>
+	);
+}
+SuppressedPluginTime.propTypes = {
+	suppressedPlugin: PropTypes.shape( {
+		timestamp: PropTypes.number,
+	} ),
+};
+
+/**
+ * Renders the username of the WP user who suppressed a plugin.
+ *
+ * @todo Maybe pass this from the backend to avoid the extra apiFetch.
+ * @todo The PHP version says "you" if the suppressing user is the current user.
```

Comment:
We could simplify this to just always/only show just the `username`.

Reply:
I have an idea for what to do. I'll give it a shot.

Reply:
We also have the `UserContextProvider` that fetches from `wp/v2/users/me`, and we could easily wrap the settings screen in that state and get current user info. So far there hasn't been a reason to bring that context provider to the settings screen but it will probably happen sooner or later.

Reply:
Amended in b826c3f4bbdc103b69dba1f90f2615eb041974f2

Reply:
Very nice 👍